sermant-plugins/sermant-flowcontrol/flowcontrol-plugin/src/main/java/io/sermant/flowcontrol/DispatcherServletInterceptor.java

```diff
@@ -85,19 +113,19 @@ private Map<String, String> getHeaders(HttpServletRequest request) {
     protected final ExecuteContext doBefore(ExecuteContext context) throws Exception {
         LogUtils.printHttpRequestBeforePoint(context);
         final Object[] allArguments = context.getArguments();
-        final HttpServletRequest argument = (HttpServletRequest) allArguments[0];
+        final Object request = allArguments[0];
         final FlowControlResult result = new FlowControlResult();
-        final Optional<HttpRequestEntity> httpRequestEntity = convertToHttpEntity(argument);
+        final Optional<HttpRequestEntity> httpRequestEntity = convertToHttpEntity(request);
         if (!httpRequestEntity.isPresent()) {
             return context;
         }
         chooseHttpService().onBefore(className, httpRequestEntity.get(), result);
         if (result.isSkip()) {
             context.skip(null);
-            final HttpServletResponse response = (HttpServletResponse) allArguments[1];
+            final Object response = allArguments[1];
             if (response != null) {
-                response.setStatus(result.getResponse().getCode());
-                response.getWriter().print(result.buildResponseMsg());
+                setStatus.accept(response, result.getResponse().getCode());
+                getWriter.apply(response).print(result.buildResponseMsg());
```

Comment:
do you need to check the writer is null?

Reply:
in spring framework, writer is not null